### PR TITLE
refactor: flatten plugin documentation

### PR DESCRIPTION
**Pull request by kneal**
This PR is designed to reduce the number of pages and directories within the plugins section. This is done by collapsing directories and taking advantage of the `index.md` markdown files that exist. 

The only section that contains some new content was the example that lived within the plugins section. Some content was updated and the file moved to `usage` to follow the pattern of in place of places example esque documentation In that section 

**Old Nav:**
Screen Shot 2021-01-12 at 2 57 26 PM

**New Nav:**
Screen Shot 2021-01-12 at 2 57 35 PM
Screen Shot 2021-01-12 at 2 59 13 PM

### CODE REVIEW

File: content/plugins/_index.md [hunk 1]
```diff
@@ -8,3 +8,76 @@ menu:
   main:
     weight: 305
 ---
+
+Vela contains two plugin types pipeline and secret. Both accept configuration via environment variables but should be used within their specific location of the Yaml pipeline. Pipeline plugins are designed to be used within steps, stages, and template pipelines. Secret plugins are designed to be used within the secrets Yaml block of pipelines.
```
Comment (KellyMerrick) [inline content/plugins/_index.md]: Vela contains two plugin types: pipeline, and secret.

File: content/plugins/_index.md [hunk 2]
```diff
@@ -8,3 +8,76 @@ menu:
...
+These actions can be for any number of general tasks, including:
+
+* deploying code
+* publishing artifacts
+* sending notifications
+* much, much more...
+
+### Sample
+
+The sample we have shown is publishing an image to a registry:Pipeline plugins configuration works via environment variables that pass data from pipeline to the container at runtime.
```
Comment (KellyMerrick) [inline content/plugins/_index.md]: The sample we have shown is publishing an image to a registry. Pipeline

File: content/plugins/_index.md [hunk 3]
```diff
@@ -8,3 +8,76 @@ menu:
...
+    pull: always
++   parameters:
++     registry: index.docker.io
++     repo: index.docker.io/octocat/hello-world
+```
+
+## Secret
+
+{{% alert title="Note:" color="warning" %}}
+Secret plugins are configured with an allow list of available images via an administator on installation. To know which secret plugins are available for your Vela installation we recommend consulting your system administrators.
```
Comment (KellyMerrick) [inline content/plugins/_index.md]: To know which secret plugins are available for your Vela installation, we recommend consulting your system administrators.

**Review by KellyMerrick — CHANGES_REQUESTED** [review]
a few grammar changes

File: content/plugins/_index.md [hunk 4]
```diff
@@ -8,3 +8,76 @@ menu:
...
+
+{{% alert title="Note:" color="warning" %}}
+Secret plugins are configured with an allow list of available images via an administator on installation. To know which secret plugins are available for your Vela installation, we recommend consulting your system administrators.
+{{% /alert %}}
+
+Secret plugins are designed to be used to read secrets in volumes within the Vela workspace. When a secret plugin runs the plugin should write data to the custom Vela mount (`/vela/secrets/`) as key/value pairs. Secret plugins configuration works via environment variables that pass data from pipeline to the container at runtime. 
+
+A secret plugin works in tandem with the Vela workspace to read data from a provider and write them into an available location (`/vela/secrets`) within a pipeline.
+
+### Sample
```
Comment (matt-fevold) [inline content/plugins/_index.md:65]: sample -> example
That's what we're going with now right?

**Review by matt-fevold — DISMISSED** [review]
one small adjustment i think but LGTM 🐕

File: content/plugins/_index.md (hunk 4 above)
Comment (kneal) [reply content/plugins/_index.md:65]: Oh yep! Will change

File: content/plugins/_index.md [hunk 5]
```diff
@@ -8,3 +8,76 @@ menu:
   main:
     weight: 305
 ---
+
+Vela contains two plugin types: pipeline, and secret. Both accept configuration via environment variables but should be used within their specific location of the Yaml pipeline. Pipeline plugins are designed to be used within steps, stages, and template pipelines. Secret plugins are designed to be used within the secrets Yaml block of pipelines.
```
Comment (jbrockopp) [inline content/plugins/_index.md]: Feels nit-picky, but often when people come into "blobs" of text, they tend to stop reading or skip over it.

Wondering if we could break this out like:

```suggestion
Vela contains two plugin types:

* pipeline: designed to be used within steps, stages, and template pipelines
* secret: designed to be used within the secrets Yaml block of pipelines

Both accept configuration via environment variables but should be used within their specific location of the Yaml pipeline.
```

Thoughts?

Comment (jbrockopp) [reply content/plugins/_index.md]: FWIW:

We could even have some of the words in the above suggestion be hyperlinks to the docs for each plugin type 🤷

Comment (kneal) [reply content/plugins/_index.md]: Yeah, I'm good with that! Blocks of text can definitely be skipped.

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by matt-fevold — APPROVED** [review]

**Pull request merged**